### ISSUE TITLE
Minimum GCC/GXX requirements

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2_dev#299]
**tldr** CUDA 9.0+ for the code, but CUDA 9.2+ for the test suite under linux, due to GCC >=7

**Edited**

flamegpu2 requires `c++14` which restritcts us to CUDA 9.0+

CUDA 9.2+ support GCC 7, but CUDA 9.0 and CUDA 9.1 require GCC < 7 (6 should work, 5.4 is supported?)

*But* GCC 5 and 6 do not seem to get along with googletest or flamegpu2 (from the ubuntu repositories, g++-5 and g++6).

#### GCC 6

When using GCC 6 with CUDA 9.1, gtest builds, but the test suite does not compile due to: 

```
googletest-src/googletest/include/gtest/internal/gtest-internal.h(1173): error: no instance of function template "testing::internal::ElemFromListImpl<testing::internal::IndexSequence<I...>>::Apply [with I=<0UL>]" matches the argument list
            argument types are: (__nv_bool (*)(), __nv_bool (*)())
          detected during:
            instantiation of class "testing::internal::ElemFromList<N, T...> [with N=1UL, T=<__nv_bool, __nv_bool>]" 
(1185): here
            instantiation of class "testing::internal::FlatTupleElemBase<testing::internal::FlatTuple<T...>, I> [with T=<__nv_bool, __nv_bool>, I=1UL]" 
(1196): here
            instantiation of class "testing::internal::FlatTupleBase<testing::internal::FlatTuple<T...>, testing::internal::IndexSequence<Idx...>> [with Idx=<0UL, 1UL>, T=<__nv_bool, __nv_bool>]" 
(1214): here
            instantiation of class "testing::internal::FlatTuple<T...> [with T=<__nv_bool, __nv_bool>]" 
/home/ptheywood/code/ptheywood/flamegpu2_dev/build/tests/googletest-src/googletest/include/gtest/internal/gtest-param-util.h(798): here
            instantiation of class "testing::internal::ValueArray<Ts...> [with Ts=<__nv_bool, __nv_bool>]" 
/home/ptheywood/code/ptheywood/flamegpu2_dev/build/tests/googletest-src/googletest/include/gtest/gtest-param-test.h(360): here
```

It seems that it might be possible to make this combination work by separating test files into a .cu file and a .cpp file, where the .cpp is the only one to include gtest. I'm not sure that this is practical however.

With GCC6, examples do build and run, it's just gtest that has issues. Not sure if we should list GCC 6 (and therefore CUDA 9&9.1 as partially supported or not)

#### GCC5

When using GCC 5, specifically 5.5 which ships with ubuntu 1604) the library cannot be built due to:

```
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx512fintrin.h(9220): error: argument of type "const void *" is incompatible with parameter of type "const float *"

/usr/lib/gcc/x86_64-linux-gnu/5/include/avx512fintrin.h(9231): error: argument of type "const void *" is incompatible with parameter of type "const float *"
```

This seems to be a common issue with gcc 5.5 when building with optimisations enabled.

**Comment by Robadob** [comment]
Readme should probably specify it too. Currently defers to CUDA gcc
requirements.
